tests/test_linalg.py

```diff
@@ -176,5 +176,5 @@ def test_backward_substitute(self, test_case: LinalgTestCase) -> None:
         mat_eliminated, rhs_eliminated, _, _ = mat.forward_eliminate(rhs_input)
         x, kernel = mat_eliminated.backward_substitute(rhs_eliminated)
         if x is not None:
-            assert np.all(x == x)
+            assert np.all(x == x)  # noqa: PLR0124 (necessary)
```

Comment:
Could we replace that with: `assert not np.any(np.isnan(x))`, which looks clearer?

Reply:
I'm afraid not.
As far as I tested, it leads to `TypeError` because `x` can sometimes be `object`-valued and thus we cannot cast the elements into `float`.

Reply:
@thierry-martinez yes, perhaps good to leave this part untouched, and think about this when `sympy` will be removed from dependency and `gflow` module refactored.